assets/src/edit-story/elements/media/edit.js

```diff
@@ -107,25 +109,62 @@ function MediaEdit({ element, box }) {
   const [croppedMedia, setCroppedMedia] = useState(null);
   const [cropBox, setCropBox] = useState(null);
   const elementRef = useRef();
+  const scaleRef = useRef(null);
+  const [updatedProperties, setUpdatedProperties] = useState({});
+  const lastUpdatedProperties = useRef({});
 
+  const setProperties = useCallback(
+    (properties) => {
+      const newProps = {
+        ...lastUpdatedProperties.current,
+        ...properties,
+      };
+      lastUpdatedProperties.current = newProps;
+      setUpdatedProperties(lastUpdatedProperties.current);
+    },
+    [setUpdatedProperties]
+  );
+
+  // Actually update the properties of the current element on unmount
   const { updateElementById } = useStory((state) => ({
     updateElementById: state.actions.updateElementById,
   }));
-  const setProperties = useCallback(
-    (properties) => updateElementById({ elementId: id, properties }),
-    [id, updateElementById]
-  );
+  const updateActualProperties = useCallback(() => {
```

Comment:
The naming `actualProperties` is a bit confusing to read -- I assumed at first that `actualProperties` would be the current actual properties that are applied to the media, however, the actual properties are the globally saved ones, right?

Could we perhaps name the local properties to `localProperties` and keep the global properties named just `properties` (and `setProperties`)? WDYT?

Reply:
Fixed in 3187660cc.